deep_quoridor/src/quoridor_env.py

```diff
@@ -232,15 +232,13 @@ def _is_wall_overlap(self, row, col, orientation):
         if self.walls[row, col, (orientation + 1) % 2] == 1:
             return True
 
-        # Check overlap with a wall in the same orientation before or after
-        deltarow = (orientation + 1) % 2
-        deltacol = orientation
-        if (
-            self.walls[min(row + deltarow, self.wall_size - 1), min(col + deltacol, self.wall_size - 1), orientation]
-            == 1
-            or self.walls[max(row - deltarow, 0), max(col - deltacol, 0), orientation] == 1
-        ):
```

Comment:
it was missing `or self.walls[row, col, orientation] == 1`, but I find this other way easier to reason about (basically check if in the 2 places where we would put segments there's a wall already, which also doesn't need to check for boundaries).

Reply:
Off the top of my head, I think this is incomplete, because it doesn't contemplate the case where you have a perpendicular wall going through in the middle of the two spaces. 
Let me see if I can prove this with a unit test, when I have a chance. In that case, you'll get -1 point :)